### PR TITLE
Support KHR_technique_webgl extension in GLTF2Loader

### PR DESCRIPTION
**Pull request by takahirox**
This PR lets `GLTF2Loader` support KHR_technique_webgl extension

Update:
I've left the existing `json.techniques` code as is so far though
`techniques` is moved to extension in glTF 2.0 because
it seems there're many models which have `techniques` under core`json`.

I'm gonna remove it when glTF 2.0 release is officially announced.

### CODE REVIEW

File: examples/js/loaders/GLTF2Loader.js [hunk 1]
```diff
@@ -1642,11 +1668,16 @@ THREE.GLTF2Loader = ( function () {
 
 					materialType = DeferredShaderMaterial;
 
-					var technique = json.techniques[ material.technique ];
+					var extension = extensions[ EXTENSIONS.KHR_TECHNIQUE_WEBGL ];
```
Comment (donmccurdy) [inline examples/js/loaders/GLTF2Loader.js]: Similar to spec-gloss, could we do something like this to factor most of this code out of this large `loadMaterials` method?

```js

materialType = DeferredShaderMaterial;
extensions[ EXTENSIONS.KHR_TECHNIQUE_WEBGL ].extendParams( materialParams, material, dependencies );

```

**Review by donmccurdy — CHANGES_REQUESTED** [review]
Nice! 👍

File: examples/js/loaders/GLTF2Loader.js (hunk 1 above)
Comment (takahirox) [inline examples/js/loaders/GLTF2Loader.js]: I'm thinking to do that when I remove the existing `json.target` code (see the Update in the PR comment)
So, adding some comments so far?

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**